.gitlab-ci.yml

```diff
@@ -0,0 +1,49 @@
+variables:
+  CI_REGISTRY: docker.io
+  CI_REGISTRY_IMAGE: jibrelnetwork/$CI_PROJECT_NAME
+docker-build-master:
+  image: docker:19.03.8
+  stage: build
+  services:
+    - docker:19.03.8-dind
+  before_script:
+    - apk add git
+    - 'which ssh-agent || ( apt-get update -y && apt-get install openssh-client -y )'
+    - eval $(ssh-agent -s)
+    - echo "$SSH_PRIVATE_KEY" | tr -d '\r' | ssh-add -
+    - mkdir -p ~/.ssh
+    - chmod 700 ~/.ssh
+    - '[[ -f /.dockerenv ]] && echo -e "Host *\n\tStrictHostKeyChecking no\n\n" > ~/.ssh/config'
+    - docker login -u "$CI_REGISTRY_USER" -p "$CI_REGISTRY_PASSWORD" $CI_REGISTRY
+
+  script:
+    - git submodule sync --recursive
+    - git submodule update --init --recursive
+    - docker build --pull -t "$CI_REGISTRY_IMAGE:latest" .
+    - docker push "$CI_REGISTRY_IMAGE:latest"
+  only:
+    - master
+
+docker-build:
+  image: docker:19.03.8
+  stage: build
+  services:
+    - docker:19.03.8-dind
+  before_script:
+    - apk add git
+    - 'which ssh-agent || ( apt-get update -y && apt-get install openssh-client -y )'
+    - eval $(ssh-agent -s)
+    - echo "$SSH_PRIVATE_KEY" | tr -d '\r' | ssh-add -
+    - mkdir -p ~/.ssh
+    - chmod 700 ~/.ssh
+    - '[[ -f /.dockerenv ]] && echo -e "Host *\n\tStrictHostKeyChecking no\n\n" > ~/.ssh/config'
+    - docker login -u "$CI_REGISTRY_USER" -p "$CI_REGISTRY_PASSWORD" $CI_REGISTRY
+
+  script:
+    - git submodule sync --recursive
```

Comment:
In most cases, the `sync` command isn't needed, because it was made for local development and downloaded repository already has a proper configuration.